crates/cheatnet/src/state.rs

```diff
@@ -236,6 +237,8 @@ pub struct CheatnetState {
     pub block_info: CheatnetBlockInfo,
     // execution resources used by all contract calls
     pub used_resources: UsedResources,
+    // trace info of the last call
+    pub trace_info: Vec<CallEntryPoint>,
```

Comment:
Couldn't we collect traces of all calls?

Reply:
And maybe create some spying interface?

Reply:
Traces of all calls - https://github.com/foundry-rs/starknet-foundry/pull/1411#discussion_r1434298661 + #1436
We can consider implementing a spying interface for trace info too